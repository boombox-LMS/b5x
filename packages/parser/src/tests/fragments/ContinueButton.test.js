const {
  exampleContinueButtonMarkup,
} = require("../../../dist/componentBuilders/fragments/ContinueButton");
const { RawFragmentSchema } = require("@b5x/types");
const { z } = require("zod");

describe("The ContinueButton fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(
    exampleContinueButtonMarkup
  );
  const rawFragment = fragment.packageForApi();

  test("The parsed continue-button fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed continue-button fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });

  test("The parsed continue-button fragment uses the expected data schema", () => {
    const validator = () => {
      z.object({}).parse(rawFragment.data);
    };
    expect(validator).not.toThrowError();
  });
});
