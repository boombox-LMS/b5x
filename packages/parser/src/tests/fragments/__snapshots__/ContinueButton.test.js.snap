// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ContinueButton fragment builds as expected The parsed continue-button fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
  ],
  "contentType": "ContinueButton",
  "contents": "",
  "data": Object {},
  "dependencyUris": Array [],
  "digest": "700437dc9a5218b71e327d06b97fa1d2",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": true,
  "isStateful": true,
  "uri": "test-fragment-uri",
}
`;
